packages/core/core/src/RequestTracker.js

```diff
@@ -527,44 +563,63 @@ export class RequestGraph extends Graph<
     // Find the `file_name` node for the parent directory and
     // recursively invalidate connected requests as described above.
     let basename = path.basename(dirname);
-    let parent = this.getNode('file_name:' + basename);
-    if (parent != null && this.hasEdge(node.id, parent.id, 'dirname')) {
-      invariant(parent.type === 'file_name');
-      this.invalidateFileNameNode(parent, dirname, matchNodes);
+    if (this.hasContentKey('file_name:' + basename)) {
+      if (
+        this.hasEdge(
+          nodeId,
+          this.getNodeIdByContentKey('file_name:' + basename),
+          'dirname',
+        )
+      ) {
+        let parent = nullthrows(
+          this.getNodeByContentKey('file_name:' + basename),
+        );
+        invariant(parent.type === 'file_name');
+        this.invalidateFileNameNode(parent, dirname, matchNodes);
```

Comment:
nit: put `'file_name:' + basename` into a variable (similar to how it was specified just once previously)